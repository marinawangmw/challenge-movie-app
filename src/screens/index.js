export { Home } from '@/screens/Home/Home';
export { Login } from '@/screens/Login/Login';
export { Profile } from '@/screens/Profile/Profile';
export { Loading } from '@/screens/Loading/Loading';
export { MovieCollection } from '@/screens/MovieCollection/MovieCollection';
export { MovieDetail } from '@/screens/MovieDetail/MovieDetail';
