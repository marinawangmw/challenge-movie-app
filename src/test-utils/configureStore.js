import { applyMiddleware, createStore } from 'redux';
import thunk from 'redux-thunk';
import { rootReducer } from '@/reducers';

const initialStore = {
  error: {},
  status: {},
  user: {},
  movieList: {
    errorMessage: undefined,
    myList: [],
    heroPoster: {},
    movieLists: [],
    similarMovies: [],
    searchResults: [],
  },
};

export function configureStore(initialState = initialStore) {
  return createStore(rootReducer, initialState, applyMiddleware(thunk));
}
