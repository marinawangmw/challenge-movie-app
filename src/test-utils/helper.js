export const movieList = {
  movieList: [
    {
      adult: false,
      backdrop_path: '/ziTxBaF1AhbUwcv2IBiFJAf54qD.jpg',
      id: 65086,
      overview: 'The story follows a young lawyer, Arthur Kipps.',
      popularity: 16.334,
      poster_path: '/aJbeDPUwDe9MLpseTFrNTBaQPzf.jpg',
      release_date: '2012-02-03',
      title: 'The Woman in Black',
      vote_average: 6.11,
    },
    {
      adult: false,
      id: 102899,
      overview: 'Armed with the astonishing ability to shrink in scale but increase in strength.',
      popularity: 69.704,
      poster_path: '/rS97hUJ1otKTTripGwQ0ujbuIri.jpg',
      release_date: '2015-07-14',
      title: 'Ant-Man',
      vote_average: 7.088,
    },
  ],
  title: 'test',
};

export const navigation = { navigate: jest.fn(), getParam: jest.fn(), setOptions: jest.fn() };
