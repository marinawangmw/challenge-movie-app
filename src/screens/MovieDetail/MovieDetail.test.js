import React from 'react';
import { render } from '@testing-library/react-native';
import { MovieDetail } from '@/screens';
import { withProviders } from '@/test-utils';

const fakeStoreWithData = {
  error: {},
  status: {},
  user: {},
  movieList: {
    errorMessage: undefined,
    myList: [],
    heroPoster: {},
    movieLists: [],
    similarMovies: {
      page: 1,
      results: [
        {
          adult: false,
          backdrop_path: '/ziTxBaF1AhbUwcv2IBiFJAf54qD.jpg',
          id: 65086,
          overview: 'The story follows a young lawyer, Arthur Kipps.',
          popularity: 16.334,
          poster_path: '/aJbeDPUwDe9MLpseTFrNTBaQPzf.jpg',
          release_date: '2012-02-03',
          title: 'The Woman in Black',
          vote_average: 6.11,
        },
        {
          adult: false,
          id: 102899,
          overview:
            'Armed with the astonishing ability to shrink in scale but increase in strength.',
          popularity: 69.704,
          poster_path: '/rS97hUJ1otKTTripGwQ0ujbuIri.jpg',
          release_date: '2015-07-14',
          title: 'Ant-Man',
          vote_average: 7.088,
        },
      ],
    },
    searchResults: [],
  },
};

const navigation = { navigate: jest.fn(), getParam: jest.fn(), setOptions: jest.fn() };

const emptyRoute = {
  params: {
    movieDetails: {
      id: null,
      poster_path: '',
      title: 'test',
      vote_average: null,
      adult: undefined,
      release_date: null,
      overview: '',
    },
  },
};

const routeWithData = {
  params: {
    movieDetails: {
      id: 592350,
      poster_path: '/zGVbrulkupqpbwgiNedkJPyQum4.jpg',
      title: 'test',
      vote_average: 8.5,
      adult: false,
      release_date: '2019-12-20',
      overview: 'Class 1-A visits Nabu Island where they finally get to do some real hero work. ',
    },
  },
};

describe('Movie Collection', () => {
  it('renders correcly with empty details and similar movies', () => {
    const { toJSON } = render(
      withProviders(<MovieDetail route={emptyRoute} navigation={navigation} />)
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders correcly with data on details and similar movies', () => {
    const { toJSON } = render(
      withProviders(<MovieDetail route={routeWithData} navigation={navigation} />, {
        initialState: fakeStoreWithData,
      })
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
