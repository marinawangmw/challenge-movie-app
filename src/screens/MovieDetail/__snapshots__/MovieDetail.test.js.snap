// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controls component renders correctly and match snapshot 1`] = `
<View
  style={
    Array [
      undefined,
      Object {
        "color": "rgb(28, 28, 30)",
      },
    ]
  }
>
  <View
    testID="controlItem"
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Image
        accessibilityIgnoresInvertColors={true}
        resizeMode="cover"
        source={null}
        style={
          Array [
            Object {
              "height": 28,
              "margin": 8,
              "width": 28,
            },
            Object {
              "height": 34,
              "marginBottom": 2,
              "marginTop": 8,
            },
          ]
        }
      />
    </View>
    <Text
      style={
        Array [
          Object {
            "color": "rgb(28, 28, 30)",
          },
          Object {
            "fontSize": 16,
            "fontWeight": "700",
          },
        ]
      }
    >
      Play
    </Text>
  </View>
</View>
`;

exports[`Message Banner component renders correctly 1`] = `
<View>
  <Text
    style={
      Object {
        "color": "rgb(28, 28, 30)",
      }
    }
  >
    There's no movie added to My List yet
  </Text>
</View>
`;

exports[`Movie Collection renders correcly with data on details and similar movies 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flex": 1,
        "height": "110%",
        "justifyContent": "center",
        "position": "absolute",
        "width": "100%",
      },
      Object {
        "backgroundColor": "rgb(242, 242, 242)CC",
      },
    ]
  }
>
  <ActivityIndicator
    animating={true}
    color="#999999"
    hidesWhenStopped={true}
    size="large"
  />
  <Text
    testID="name"
  >
    Hello, undefined!
  </Text>
</View>
`;

exports[`Movie Collection renders correcly with empty details and similar movies 1`] = `null`;
